Common/src/Storage/SessionTableExtensions.cs

```diff
@@ -110,4 +112,169 @@ public static class SessionTableExtensions
+                                                           CancellationToken  cancellationToken = default)
+    => await sessionTable.UpdateOneSessionAsync(sessionId,
+                                                data => data.Status == SessionStatus.Paused,
+                                                new List<(Expression<Func<SessionData, object?>> selector, object? newValue)>
+                                                {
+                                                  (model => model.Status, SessionStatus.Running),
+                                                },
+                                                false,
+                                                cancellationToken)
+                         .ConfigureAwait(false) ?? throw new SessionNotFoundException($"No paused session with {sessionId} found.");
+
+  /// <summary>
+  ///   Purge a session
+  /// </summary>
+  /// <param name="sessionTable">Interface to manage sessions lifecycle</param>
+  /// <param name="sessionId">Id of the session to purge</param>
+  /// <param name="cancellationToken">Token used to cancel the execution of the method</param>
+  /// <returns>
+  ///   The metadata of the purged session
+  /// </returns>
+  public static async Task<SessionData> PurgeSessionAsync(this ISessionTable sessionTable,
+                                                          string             sessionId,
+                                                          CancellationToken  cancellationToken = default)
+    => await sessionTable.UpdateOneSessionAsync(sessionId,
+                                                data => data.Status == SessionStatus.Running || data.Status == SessionStatus.Paused,
+                                                new List<(Expression<Func<SessionData, object?>> selector, object? newValue)>
+                                                {
+                                                  (model => model.Status, SessionStatus.Purged),
+                                                  (model => model.PurgeDate, DateTime.UtcNow),
+                                                },
+                                                false,
+                                                cancellationToken)
+                         .ConfigureAwait(false) ?? throw new SessionNotFoundException($"No open session with {sessionId} found.");
+
+  /// <summary>
+  ///   Delete a session
+  /// </summary>
+  /// <param name="sessionTable">Interface to manage sessions lifecycle</param>
+  /// <param name="sessionId">Id of the session to delete</param>
+  /// <param name="cancellationToken">Token used to cancel the execution of the method</param>
+  /// <returns>
+  ///   The metadata of the deleted session
+  /// </returns>
+  /// <remarks>
+  ///   A session cannot be deleted twice
+  /// </remarks>
+  public static async Task<SessionData> DeleteSessionAsync(this ISessionTable sessionTable,
```

Comment:
Should Delete also purge the session? Would we need the Delete call to also stop submission?

Reply:
We may want to cancel running tasks beforehand